lib/widgets/poll.dart

```diff
@@ -69,48 +84,74 @@ class _PollWidgetState extends State<PollWidget> {
+                // than the vote count box.  Still, we avoid padding at the
+                // start because we want to align all the poll options to the
+                // surrounding messages.
+                padding: const EdgeInsetsDirectional.only(
+                  end: 5, top: verticalPadding, bottom: verticalPadding),
+                child: Container(
+                  // Inner padding preserves whitespace even when the text's
+                  // width approaches the button's min-width (e.g. because
+                  // there are more than three digits).
+                  padding: const EdgeInsets.symmetric(horizontal: 4),
+                  decoration: BoxDecoration(
+                    color: theme.colorPollVoteCountBackground,
+                    border: Border.all(color: theme.colorPollVoteCountBorder),
+                    borderRadius: BorderRadius.circular(3)),
+                  child: Center(
+                    child: Text(option.voters.length.toString(),
+                      style: textStyleBold.copyWith(
+                        color: theme.colorPollVoteCountText, fontSize: 20))))))),
+          Expanded(
+            child: Padding(
+              // This and the padding on the vote count box both extend the row
+              // by the same extent. This ensures that there still will be
+              // consistent spacing between rows when the text takes more
+              // vertical space than the vote count box.
+              padding: const EdgeInsets.symmetric(vertical: verticalPadding),
               child: Wrap(
                 spacing: 5,
                 children: [
                   Text(option.text, style: textStyleBold.copyWith(fontSize: 16)),
                   if (option.voters.isNotEmpty)
                     // TODO(i18n): Localize parenthesis characters.
                     Text('($voterNames)', style: textStyleVoterNames),
-                ])),
-          ]));
+                ]))),
+        ]);
     }
 
-    return Column(
-      crossAxisAlignment: CrossAxisAlignment.start,
-      children: [
-        Padding(padding: const EdgeInsets.only(bottom: 6), child: question),
-        if (widget.poll.options.isEmpty)
-          Text(zulipLocalizations.pollWidgetOptionsMissing,
-            style: textStyleVoterNames.copyWith(fontStyle: FontStyle.italic)),
-        for (final option in widget.poll.options)
-          buildOptionItem(option),
-      ]);
+    return Padding(
+      // `verticalPadding` out of 5 logical pixels
+      // come from the last option row.
+      padding: const EdgeInsets.only(bottom: 5 - verticalPadding),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Padding(
+            // `verticalPadding` out of 6 logical pixels
+            // come from the first option row.
+            padding: const EdgeInsets.only(bottom: 6 - verticalPadding),
+            child: question),
```

Comment:
nit: this is a bit simpler to express with a SizedBox spacer:
```suggestion
          question,
          // `verticalPadding` out of 6px come from the first option row.
          const SizedBox(height: 6 - verticalPadding),
```